File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        size = len(nums)
        result = [0] * size
        prefix = [0] * size
        suffix = [0] * size

        prefix[0] = suffix[size - 1] = 1
        for i in range(1, size): 
            prefix[i] = nums[i - 1] * prefix[i - 1]
        for i in range(size - 2, -1, -1): 
            suffix[i] = nums[i + 1] * suffix[i + 1]
        for i in range(size): 
            result[i] = prefix[i] * suffix[i]
        return result

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const mp = new Map();
        let result = 0;

        for (let num of nums) {
            if (!mp.has(num)) {
                mp.set(
                    num,
                    (mp.get(num - 1) || 0) + (mp.get(num + 1) || 0) + 1, 
                );
                mp.set(num - (mp.get(num - 1) || 0), mp.get(num));
                mp.set(num + (mp.get(num + 1) || 0), mp.get(num));
                result = Math.max(result, mp.get(num));
            }
        }
        return result;
    }
}
